airdrop-machine/sources/airdrop_machine.move

```diff
@@ -278,9 +302,41 @@ module airdrop_machine::airdrop_machine {
         borrow_global_mut<CollectionConfig>(object::object_address(&collection_config_object))
     }
 
-    fun get_pseudo_random_index(length: u64): u64 {
+    fun get_weighted_pseudo_random_index(token_uris_weights: &vector<u64>): u64 {
+        let total_weight = 0u64;
+        let len = vector::length(token_uris_weights);
+
+        // Compute the total weight
+        let i = 0;
+        while (i < len) {
+            total_weight = total_weight + *vector::borrow(token_uris_weights, i);
+            i = i + 1;
+        };
```

Comment:
We can replace all this with

`let total_weight = vector::fold(*token_uris_weights, 0, |l, r| l + r);`